fixture-monkey-api/src/main/java/com/navercorp/fixturemonkey/api/arbitrary/StringCombinableArbitrary.java

```diff
@@ -0,0 +1,119 @@
+ * limitations under the License.
+ */
+
+package com.navercorp.fixturemonkey.api.arbitrary;
+
+import java.util.function.Predicate;
+import java.util.regex.Pattern;
+
+public interface StringCombinableArbitrary extends CombinableArbitrary<String> {
+	int STRING_DEFAULT_MIN_LENGTH = 0;
+	int STRING_DEFAULT_MAX_LENGTH = 100;
+
+	@Override
+	String combined();
+
+	@Override
+	String rawValue();
+
+	StringCombinableArbitrary withLength(int min, int max);
+
+	/**
+	 * Generates a StringCombinableArbitrary which contains only alphabetic characters.
+	 * It conflicts with {@link #ascii()} and {@link #numeric()} and {@link #korean()}.
+	 *
+	 * @return the StringCombinableArbitrary which contains only alphabetic characters
+	 */
+	StringCombinableArbitrary alphabetic();
+
+	/**
+	 * Generates a StringCombinableArbitrary which contains only ASCII characters.
+	 * It conflicts with {@link #alphabetic()} and {@link #numeric()} and {@link #korean()}.
+	 *
+	 * @return the StringCombinableArbitrary which contains only ASCII characters
+	 */
+	StringCombinableArbitrary ascii();
+
+	/**
+	 * Generates a StringCombinableArbitrary which contains only numeric characters.
+	 * It conflicts with {@link #alphabetic()} and {@link #ascii()} and {@link #korean()}.
+	 *
+	 * @return the StringCombinableArbitrary which contains only numeric characters
+	 */
+	StringCombinableArbitrary numeric();
+
+	/**
+	 * Generates a StringCombinableArbitrary which contains only Korean characters.
+	 * It conflicts with {@link #alphabetic()} and {@link #ascii()} and {@link #numeric()}.
+	 *
+	 * @return the StringCombinableArbitrary which contains only Korean characters
+	 */
+	StringCombinableArbitrary korean();
+
+	default StringCombinableArbitrary withMinLength(int min) {
+		return this.withLength(min, STRING_DEFAULT_MAX_LENGTH);
+	}
+
+	default StringCombinableArbitrary withMaxLength(int max) {
+		return this.withLength(STRING_DEFAULT_MIN_LENGTH, max);
+	}
+
+	@Override
+	default StringCombinableArbitrary filter(Predicate<String> predicate) {
+		return this.filter(DEFAULT_MAX_TRIES, predicate);
+	}
+
+	@Override
+	default StringCombinableArbitrary filter(int tries, Predicate<String> predicate) {
+		return new StringCombinableArbitraryDelegator(CombinableArbitrary.super.filter(tries, predicate));
+	}
+
+	default StringCombinableArbitrary filterCharacter(Predicate<Character> predicate) {
+		return this.filterCharacter(DEFAULT_MAX_TRIES, predicate);
+	}
+
+	StringCombinableArbitrary filterCharacter(int tries, Predicate<Character> predicate);
+
+	/**
+	 * Generates the pattern matched arbitrary String.
+	 * Its performance differs from the generation engine you're using.
+	 *
+	 * @param stringPattern the regular expression pattern
+	 * @return the pattern matched StringCombinableArbitrary
+	 */
+	default StringCombinableArbitrary pattern(int tries, String stringPattern) {
+		Pattern pattern = Pattern.compile(stringPattern);
```

Comment:
How about adding method taking `Pattern` class? It seems [immutable](https://github.com/openjdk/jdk/blob/c873837da6e373613866f5f5c0017f0fccb97b57/src/java.base/share/classes/java/util/regex/Pattern.java#L77)